archimedes/registry.py

```diff
@@ -0,0 +1,192 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#   Valerio Cosentino <valcos@bitergia.com>
+#
+
+import json
+import logging
+import os
+
+from archimedes.errors import NotFoundError, RegistryError
+from archimedes.meta_obj import MetaObj
+from archimedes.utils import load_json
+
+logger = logging.getLogger(__name__)
+
+REGISTRY_NAME = ".registry"
+
+
+class Registry:
+
+    def __init__(self, root_path):
+        """This class allows to manipulate the Kibana objects using aliases, thus avoiding the user
+        to deal with their IDs and titles. The registry is saved in the root folder of Archimedes,
+        with name .registry. The content of the registry is as following:
+        [
+            "1": {
+                    'id': 'Search:_pull_request:false',
+                    'title': 'Search:_pull_request:false',
+                    'type': 'search',
+                    'version': 1
+            },
+            "2": {
+                    'id': '8539ada0-9960-11e8-8771-a349686d998a',
+                    'title': 'dockerhub',
+                    'type': 'index-pattern',
+                    'version': 1
+            },
+            ...
+        ]
+
+        :param root_path: path where the registry will be stored
+        """
+        self.path = os.path.join(root_path, REGISTRY_NAME)
+
+        if not os.path.exists(self.path):
+            self.__create_registry()
+
+    def find(self, alias=None, obj_type=None):
+        """Return the information in the registry. If `alias` and `obj_type` are null, it returns the
+        content of all the registry. Otherwise, it returns the information related to the single `alias` or
+        the aliases with the given `obj_type`.
+
+        :param alias: target alias
+        :param obj_type: target object type
+
+        :returns the information related to the alias
+        """
+        registry = load_json(self.path)
+
+        if alias:
+            if alias not in registry:
+                cause = "Alias %s not found in registry" % alias
+                logger.error(cause)
+                raise NotFoundError(cause=cause)
+
+            meta = MetaObj.create_from_registry(registry[alias])
+            yield alias, meta
+
+        elif obj_type:
+            for alias in registry.keys():
+                if registry[alias]['type'] == obj_type:
+                    meta = MetaObj.create_from_registry(registry[alias])
+                    yield alias, meta
+        else:
+            for alias in registry.keys():
+                meta = MetaObj.create_from_registry(registry[alias])
+                yield alias, meta
+
+    def delete(self, alias=None):
```

Comment:
Probably two methods will be fine here. One `clear` to empty the registry and another `delete` to remove by alias.